libr/util/axml.c

```diff
@@ -0,0 +1,281 @@
+/* radare2 - LGPL - Copyright 2021 - keegan */
+
+#include <string.h>
+#include <r_types.h>
+#include <r_core.h>
+
+enum {
+	TYPE_STRING_POOL = 0x0001,
+	TYPE_XML = 0x0003,
+	TYPE_START_ELEMENT = 0x0102,
+	TYPE_END_ELEMENT = 0x103,
+};
+
+enum {
+	RESOURCE_STRING = 0x03,
+};
+
+// Beginning of every header
+R_PACKED (
+typedef struct {
+	ut16 type;
+	ut16 header_size;
+	ut32 size;
+}) chunk_header_t;
+
+// String pool referenced throughout the Binary XML, there must only be ONE
+R_PACKED (
+typedef struct {
+	ut32 string_count;
+	ut32 style_count;
+	ut32 flags;
+	ut32 strings_offset;
+	ut32 styles_offset;
+	ut32 offsets[];
+}) string_pool_t;
+
+R_PACKED (
+typedef struct {
+	ut16 size;
+	ut8 unused;
+	ut8 type;
+	ut32 data;
+}) resource_value_t;
+
+R_PACKED (
+typedef struct {
+	ut32 namespace;
+	ut32 name;
+	ut32 unused;
+	resource_value_t value;
+}) attribute_t;
+
+R_PACKED (
+typedef struct {
+	ut32 line;
+	ut32 comment;
+	ut32 namespace;
+	ut32 name;
+	ut32 flags;
+	ut16 attribute_count;
+	char unused[sizeof(ut16) * 3];
+	attribute_t attributes[];
```

Comment:
uhm 0 size fields in structs can be problematic and cause programmer errors if the allocation of the struct field is not in sync. ive removed them from other parts of r2 to avoid such issues to happen, if you can change that logic easily i would prefer it.